app/controllers/teachers/lessons_controller.rb

```diff
@@ -1,55 +1,13 @@
 module Teachers
   class LessonsController < BaseController
     def show
-      @lesson = Lesson.find(params[:id])
+      @lesson = Lesson.eager_load(lesson_parts: :activities).find(params[:id])
 
-      # lifted from prototype, these fields should guide
-      # the building of the LessonPart model
-      @lesson_parts = [
-        OpenStruct.new(
-          number: 1,
-          activity_type: 'Practical',
-          involvement: 'Whole class',
-          overview: <<~OVERVIEW,
-            Pupils demonstrate independent thinking, applying their prior
-            knowledge of volcanoes and focussing on the effects of volcanic
-            eruptions.
-
-            Ask pupils to watch the video carefully, noting as many words as
-            they can about what they observe. Remind them of their last lesson
-            to help them.
-
-            Follow up with a discussion to assess answers and vocabulary. Use
-            further questioning to ask pupils for an example of a
-            socio-economic impact. What could happen as a result of what they
-            have observed in the video? The eruption may result in lava flows
-            and ash, for example, making visibility poor and leaving people
-            injured.
-          OVERVIEW
-          duration: '10 minutes',
-          file_formats: %w(.pdf .doc .mov),
-          tags: ['Extra prep'],
-          alternatives: ['Pair reading available', 'Worksheet available'],
-          extra_requirements: ['Extra prep']
-        ),
-        OpenStruct.new(
-          number: 2,
-          activity_type: 'Practical',
-          involvement: 'Whole class',
-          overview: <<~OVERVIEW,
-            Both pupils read an activity sheet explaining the differences
-            between active, dormant, and extinct volcanoes.
-
-            Together they should work to correctly label the images of the
-            different volcanoes presented.
-          OVERVIEW
-          duration: '10 minutes',
-          file_formats: %w(.pdf .doc .mov),
-          tags: ['Extra prep'],
-          alternatives: ['Pair reading available', 'Worksheet available'],
-          extra_requirements: ['Extra prep']
-        )
-      ]
+      # We only want the 'active' activities here; those that have been
+      # selected by the current teacher or, if no selection has been made, the
+      # default.
+      # FIXME optimise the query
+      @activities = @lesson.lesson_parts.map { |lp| lp.activity_for(current_teacher) }
```

Comment:
I'd been playing around with this (even came up with the same method name) but I'm not sure if we want the activity_for method on the lesson_part, on the teacher, or whether this is actually just a ui concern and we should have some sort of presenter (ActivityList ?).
I wonder if we should create an ActivityChoice for the teacher from the default activity before they first visit this page then we don't have to check if the activity is chosen or is the default, however I can't see a good place to do this in the code.

Reply:
It _could_ be a UI concern, but more generally it's just telling us which activity a teacher would see if they viewed the lesson. I think this can be simplified further but as it's not touching any data will refactor this later I think